app/controllers/presenters_controller.rb

```diff
@@ -9,4 +9,24 @@ def new
   def create
   end
 
+  # list current profiles and provide CrUD like interface. 
+  def edit_subjects
+  	@presenter = current_user.presenter
+  end
+
+  # add a subject to the presenter's subject
+  def add_subject
+  	@subject = Subject.find_by(name: params[:subjects][:name])
+  	unless current_user.presenter.subjects.include?(@subject)
+      current_user.presenter.subjects << @subject 
+    end
+  	redirect_to presenter_edit_subjects_path
+  end
+
+  def remove_subject
+    @subject = Subject.find(params[:subject])
+    @subject.presenters.delete(current_user.presenter)
```

Comment:
REMOVE, not delete. 
Only an admin can delete a subject. `remove_subject` removes the subject from the presenter's subjects.